Django 3 support.

heydenberk opened the pull request:
Closes #354.

Django 3 ["remove[s] private Python 2 compatibility APIs"](https://docs.djangoproject.com/en/3.0/releases/3.0/#removed-private-python-2-compatibility-apis), which is to say it removes a number of utility methods that were either already provided by `six` or otherwise were used in bridging the old and the new python.  This pull request makes `django-waffle` work with Django 3 by making just 3 changes.

1. Add `six` to the requirements.
2. Replace the result of calling`available_attrs` with `functools.WRAPPER_ASSIGNMENTS` (cf https://docs.djangoproject.com/en/2.2/_modules/django/utils/decorators/)
3. Use the `python_2_unicode_compatible` provided by `six` rather than the one vendored with django.

Not sure what the protocol is on bumping the version and amending the changelog; would be happy to do both.

heydenberk committed:
Add explict dependency on six.

heydenberk committed:
Replace available_attrs(arg) with WRAPPER_ASSIGNMENTS.

heydenberk committed:
Import python_2_unicode_compatible from six rather than django.

heydenberk committed:
Add six to install_requires.

clintonb reviewed (commented):
Thanks for taking the initiative. Here is what I'd like to see before merging:

- [x] Update supported Django version in the `classifiers` list in `setup.py`
- [x] Update `tox.ini` to runt tests with Django 3.0
- [x] Add an entry to `CHANGES` for v0.19.0

heydenberk committed:
Add Django 3.0 to tox.ini.

heydenberk committed:
Add Django 3.0 to setup.py classifiers.

heydenberk committed:
Add changelog entry for v0.19

clintonb committed:
Using Travis matrix to split build jobs

We now use an explict matrix for all builds to avoid issues that arise from database connections not being properly closed between builds.
